modules/auxiliary/gather/kerberos_enumusers.rb

```diff
@@ -29,84 +34,66 @@ def initialize(info = {})
 
     register_options(
       [
-        OptString.new('DOMAIN', [ true, 'The Domain Eg: demo.local' ]),
-        OptPath.new(
-          'USER_FILE',
-          [true, 'Files containing usernames, one per line', nil]
-        )
-      ],
-      self.class
+        OptString.new('DOMAIN', [ true, 'The Domain Eg: demo.local' ])
+      ]
     )
-  end
 
-  def user_list
-    if File.readable? datastore['USER_FILE']
-      users = File.new(datastore['USER_FILE']).readlines(chomp: true)
-      users.each(&:downcase!)
-      users.uniq!
-    else
-      raise ArgumentError, "Cannot read file #{datastore['USER_FILE']}"
-    end
-    users
+    deregister_options('BLANK_PASSWORDS', 'BRUTEFORCE_SPEED', 'DB_ALL_CREDS', 'DB_ALL_PASS', 'PASSWORD', 'PASS_FILE', 'USER_AS_PASS', 'USERPASS_FILE')
   end
 
   def run
     domain = datastore['DOMAIN'].upcase
     print_status("Using domain: #{domain} - #{peer}...")
 
-    pre_auth = []
-    pre_auth << build_pa_pac_request
-    pre_auth
-
-    user_list.each do |user|
-      next if user.empty?
-
-      begin
-        res = send_request_as(
-          client_name: user.to_s,
-          server_name: "krbtgt/#{domain}",
-          realm: domain.to_s,
-          pa_data: pre_auth
-        )
-      rescue ::EOFError => e
-        print_error("#{peer} - User: #{user.inspect} - EOF Error #{e.message}. Aborting...")
-        elog(e)
-        # Stop further requests entirely
-        return false
-      rescue Rex::Proto::Kerberos::Model::Error::KerberosDecodingError => e
-        print_error("#{peer} - User: #{user.inspect} - Decoding Error -  #{e.message}. Aborting...")
-        elog(e)
-        # Stop further requests entirely
-        return false
-      end
-
-      case res.msg_type
-      when Rex::Proto::Kerberos::Model::AS_REP
-        hash = format_asrep_to_john_hash(res)
+    cred_collection = build_credential_collection({ usernames_only: true })
+    pre_auth = [build_pa_pac_request]
+    scanner = ::Metasploit::Framework::LoginScanner::Kerberos.new(
+      host: self.rhost,
+      port: self.rport,
+      server_name: "krbtgt/#{domain}",
+      realm: domain.to_s,
+      pa_data: pre_auth,
```

Comment:
Can we default this but allow overriding the value? I imagine most folk won't need to deal with the pa_data themselves 🤔